karoo_gp/base_class.py

```diff
@@ -211,68 +194,141 @@ def error(self, msg, display={'i', 'g', 'm', 'db'}):
     #   Methods to Run Karoo GP                  |
     #+++++++++++++++++++++++++++++++++++++++++++++
 
-    def fit(self, X=None, y=None):
-
-        '''
-        This method enables the engagement of the entire Karoo GP application.
-        Instead of returning the user to the pause menu, this script terminates
-        at the command-line, providing support for bash and chron job execution.
+    def fitness_compare(self, a, b):
+        """Return the fitter of two Trees, or the latter if equal"""
+        op = operator.gt if self.higher_is_better else operator.lt
+        return a if op(a.score['fitness'], b.score['fitness']) else b
```

Comment:
Good point about making `fitness_compare` overridable.

`Tree`s could still be made comparable if it makes sense, and you don't need to sort them -- you can just use `min`/`max` (taking the lowest-index is error-prone since it relies on the population being always sorted properly).  This is also something that you might be able to do here -- e.g.
```py
    def fitness_compare(self, a, b):
        """Return the fitter of two Trees, or the latter if equal"""
        fun = max if self.higher_is_better else min
        return fun(self.population)